build.cake

```diff
@@ -0,0 +1,158 @@
+//////////////////////////////////////////////////////////////////////
+
+var version = "0.4.0";
+var modifier = "";
+
+var dbgSuffix = configuration == "Debug" ? "-dbg" : "";
+var packageVersion = version + modifier + dbgSuffix;
+
+if (BuildSystem.IsRunningOnAppVeyor)
+{
+	var tag = AppVeyor.Environment.Repository.Tag;
+
+	if (tag.IsTag)
+	{
+		packageVersion = tag.Name;
+	}
+	else
+	{
+		var buildNumber = AppVeyor.Environment.Build.Number.ToString("00000");
+		var branch = AppVeyor.Environment.Repository.Branch;
+		var isPullRequest = AppVeyor.Environment.PullRequest.IsPullRequest;
+
+		if (branch == "master" && !isPullRequest)
+		{
+			packageVersion = version + "-dev-" + buildNumber + dbgSuffix;
+		}
+		else
+		{
+			var suffix = "-ci-" + buildNumber + dbgSuffix;
+
+			if (isPullRequest)
+				suffix += "-pr-" + AppVeyor.Environment.PullRequest.Number;
+			else
+				suffix += "-" + branch;
+
+			// Nuget limits "special version part" to 20 chars. Add one for the hyphen.
+			if (suffix.Length > 21)
+				suffix = suffix.Substring(0, 21);
+
+                        suffix = suffix.Replace(".", "");
+
+			packageVersion = version + suffix;
+		}
+	}
+
+	AppVeyor.UpdateBuildVersion(packageVersion);
+}
+
+var packageName = "NUnit3TestAdapter-" + packageVersion;
+
+//////////////////////////////////////////////////////////////////////
+// DEFINE RUN CONSTANTS
+//////////////////////////////////////////////////////////////////////
+
+// Directories
+var PROJECT_DIR = Context.Environment.WorkingDirectory.FullPath + "/";
+var PACKAGE_DIR = PROJECT_DIR + "package/";
+var PACKAGE_IMAGE_DIR = PACKAGE_DIR + packageName + "/";
+var TOOLS_DIR = PROJECT_DIR + "tools/";
+var BIN_DIR = PROJECT_DIR + "bin/" + configuration + "/";
+
+// Files
+var SOLUTION_FILE = PROJECT_DIR + "nunit-summary.sln";
+var TEST_ASSEMBLY = BIN_DIR + "test-nunit-summary.exe";
+
+//////////////////////////////////////////////////////////////////////
+// CLEAN
+//////////////////////////////////////////////////////////////////////
+
+Task("Clean")
+    .Does(() =>
+{
+    CleanDirectory(BIN_DIR);
+});
+
+
+//////////////////////////////////////////////////////////////////////
+// NuGet Restore
+//////////////////////////////////////////////////////////////////////
+
+Task("NuGetRestore")
+    .Does(() =>
+{
+    NuGetRestore(SOLUTION_FILE);
+});
+
+//////////////////////////////////////////////////////////////////////
+// BUILD
+//////////////////////////////////////////////////////////////////////
+
+Task("Build")
+    .IsDependentOn("NuGetRestore")
+    .Does(() =>
+    {
+		DotNetBuild(SOLUTION_FILE, settings =>
+			settings.SetConfiguration(configuration)
+					.SetVerbosity(Verbosity.Minimal)
+					.WithTarget("Build"));
+    });
+
+//////////////////////////////////////////////////////////////////////
+// TEST
+//////////////////////////////////////////////////////////////////////
+
+Task("Test")
+	.IsDependentOn("Build")
+	.Does(() =>
+	{
+		int rc = StartProcess(
```

Comment:
It would have been nice to start using the Cake NUnit task in a few of these side projects, but this works.